### PR TITLE
low storage better estimate for prepc fluxes

### PR DESCRIPTION
**Pull request by mjberger**
…new low storage prepc made listbc real and added a variable to remember the start of the space on each level. Saved it in check/rest as well.

### CODE REVIEW

File: src/2d/prepc.f [hunk 1]
```diff
@@ -26,16 +26,33 @@ subroutine prepc(level,nvar)
...
          maxsp  = maxsp + 2*(ikeep+jkeep)
       mkid = node(levelptr,mkid)
       go to 10
- 20   listsp(level) = maxsp
-      if (maxsp .eq. 0) go to 99
+ 20   continue
+ 
+      !!! maxsp is enough storage for every coarse grid to have an adjustment by
+      !!! a corresponding fine grid flux sum. But not all will.
+      if (maxsp .eq. 0) go to 99  ! no fine grids, so no space needed. CHECK HOW THIS IS MARKED FOR COARSE GRIDS
```
Comment (rjleveque) [inline src/2d/prepc.f:44]: @mjberger: there's a comment in prepc.f:
CHECK HOW THIS IS MARKED FOR COARSE GRIDS
Is this something that still needs to be checked, or delete the comment?

**Comment by mjberger** [comment]
already checked. forgot to delete. sorry.

 — Marsha

**Comment by rjleveque** [comment]
You can delete it, commit and push to this same branch and it will update the PR automatically.

**Comment by mjberger** [comment]
will do right now. Thanks, — Marsha

**Comment by mandli** [comment]
How much more space does this seem to save then in tests?

**Comment by rjleveque** [comment]
@mandli: For big GeoClaw runs it can be a factor of 20 or more.  In a test I just ran the old code died when trying to extend alloc past length approximately 2 billion (at which point integer*4 is not sufficient for indexing), while the new code ran fine, and reports that the maximum length of alloc was about 88 million.  There were at most 7-8 million grid cells at any time in this computation.

Thanks @mjberger!

**Comment by mandli** [comment]
Nice!

**Comment by mandli** [comment]
Since this is passing tests and looks fine to me I will merge this and then look at the GeoClaw changes.

**Pull request merged**